docs/tutorials/using-ev3-speaker.md

```diff
@@ -0,0 +1,196 @@
+able to reproduce a recorded sound.
+
+[1]: https://en.wikipedia.org/wiki/Audio_frequency
+
+
+## Tone Mode
+
+There are a couple of ways to control the speaker in Tone mode. Also worth an
+honorable mention: if you plug a USB keyboard into the EV3 and press
+<kbd>CTRL</kbd>+<kbd>G</kdb>, it will beep!
+
+
+### beep command
+
+The [beep man page][2] has all you need to know.
+
+[2]: https://manpages.debian.org/jessie/beep/beep.1.en.html
+
+Examples:
+
+```bash
+$ beep # 440 Hz for 200 ms
+$ beep -f 200 -l 500 # 220 Hz for 500 ms
+$ beep -f 262 -l 180 -d 20 -r 2 \
+-n -f 392 -l 180 -d 20 -r 2 \
+-n -f 440 -l 180 -d 20 -r 2 \
+-n -f 392 -l 380 -d 20 \
+-n -f 349 -l 180 -d 20 -r 2 \
+-n -f 330 -l 180 -d 20 -r 2 \
+-n -f 294 -l 180 -d 20 -r 2 \
+-n -f 262 -l 400 # twinkle twinkle little star
+```
+
+
+### tone sysfs Attribute
+
+TODO: using the `tone` attribute is not really recommended. This needs to be
+updated with an example on how to use the Linux input subsystem instead.
+
+We have also provided a way to play tones directly using the sysfs attribute
+`/sys/devices/platform/snd-legoev3/tone`.
+
+Example:
+
+```bash
+$ cd /sys/devices/platform/snd-legoev3
+$ echo 1000 10000 > tone # play a 1 KHz tone for 10 seconds
+$ echo 440 > tone # play a 440 Hz tone (endless mode)
+$ echo 0 > tone # stop the tone 
+```
+
+
+## PCM Playback Mode
+
+PCM playback is implemented as an ALSA driver. This means that you can use just
+about any Linux program out there that can play sound. We have found the best
+compromise between sound quality and CPU usage is to use a sample rate 22050 Hz.
+16000 Hz sounds pretty good too. Of course the speaker is so small, the sound
+quality is just not going to be "high quality" no matter what.
+
+Example: Playing back a .wav file using `aplay`.
+
+```bash
+$ aplay my-file.wav
+Playing WAVE 'my-file.wav' : Signed 16 bit Little Endian, Rate 22050 Hz, Mono
+```
+
+{% include /style/icon.html type="info" %}
+You must be a member of the `audio` group in order to use the sound driver.
+{: .alert .alert-info}
+
+
+## Volume control
+
+Again, there are a number of ways to do this. Currently, there is only a master
+volume control that controls both Tone mode and PCM Playback mode.
+
+### amixer Command
+
+Documentation:
+* [man page](https://manpages.debian.org/jessie/alsa-utils/amixer.1.en.html)
+
+Example:
+
+```bash
+$ amixer get Playback,0 # get current volume
+Simple mixer control 'Playback',0
+  Capabilities: volume volume-joined penum
+  Playback channels: Mono
+  Capture channels: Mono
+  Limits: 0 - 256
+  Mono: 255 [100%]
+$ amixer set Playback,0 50% # set volume to 50%
+Simple mixer control 'Playback',0
+  Capabilities: volume volume-joined penum
+  Playback channels: Mono
+  Capture channels: Mono
+  Limits: 0 - 256
+  Mono: 128 [50%]
+```
+
+### alsamixer Command
+
+Graphical volume controls.
+
+```
+┌───────────────────────────── AlsaMixer v1.0.25 ──────────────────────────────┐
```

Comment:
Hmm... most sites horizontally scroll `<pre>` regions when they are too wide. I suppose we can tweak our css for this?